src/librustc/dep_graph/visit.rs

```diff
@@ -78,9 +79,30 @@ pub fn visit_all_item_likes_in_krate<'a, 'tcx, V, F>(tcx: TyCtxt<'a, 'tcx, 'tcx>
 pub fn visit_all_bodies_in_krate<'a, 'tcx, C>(tcx: TyCtxt<'a, 'tcx, 'tcx>, callback: C)
     where C: Fn(/* body_owner */ DefId, /* body id */ hir::BodyId),
 {
+    // NB: we use a visitor here rather than walking the keys of the
+    // hashmap so as to ensure we visit the bodies "in order".
```

Comment:
Well, first off, the hashmap should maybe be replaced with a `BTreeMap`.
But also, you don't want HIR lowering node IDs to matter... what if it was keyed by the owner's `DefIndex`?